src/workerd/io/io-channels.h

```diff
@@ -146,6 +146,8 @@ class IoChannelFactory {
     // This string remains valid at least until either the request has returned response headers
     // or has been canceled. (In practice, this string's lifetime is that of the Isolate making
     // the request.)
+
+    SubrequestMetadata clone();
```

Comment:
I'm a little bit concerned about `SubrequsetMetadata` becoming non-linear, this implies that we can no longer have metadata that isn't clonable (e.g. a callback of some sort, which is actually something that came up in a different PR today).

It's not clear to me why it needs to be cloned. Shouldn't `Loopback::getWorker()` require the caller to provide a new `SubrequestMetadata`?

For the specific use case of websocket hibernation, it seems like `cfBlobJson` and `featureFlagsForFl` are not relevant, which leaves only `parentSpan`. It makes sense to me that "loopback" requests should be nested within the original request's span, not siblings of it, which suggests that indeed the caller of `Loopback::getWorker()` should be providing new metadata.

Reply:
I think this would also greatly reduce the amount of churn in `server.c++` in this PR, which seems to be mostly about handling this metadata?